operator/check_operator.go

```diff
@@ -84,11 +65,47 @@ func (c operatorCheck) Run(ctx context.Context) (certification.Results, error) {
 		return certification.Results{}, err
 	}
 
+	return eng.Results(ctx), nil
+}
+
+func (c *operatorCheck) resolve(ctx context.Context) error {
+	if c.resolved {
+		return nil
+	}
+
+	switch {
+	case c.image == "":
+		return preflighterr.ErrImageEmpty
+	case c.kubeconfig == nil:
+		return preflighterr.ErrKubeconfigEmpty
+	case c.indeximage == "":
+		return preflighterr.ErrIndexImageEmpty
+	}
+
+	c.policy = policy.PolicyOperator
+	var err error
+	c.checks, err = engine.InitializeOperatorChecks(ctx, c.policy, engine.OperatorCheckConfig{
```

Comment:
Same comment as before.  Don't set fields directly if there is an error possibility.